packages/hmr-plugin/src/internal/hmr-options-builder.ts

```diff
@@ -1,29 +1,26 @@
-import { isJIT } from '@ngxs/store/internals';
-
 import { NgxsHmrOptions } from '../symbols';
-import { ivyEnabledInDevMode } from '../../../store/src/ivy/ivy-enabled-in-dev-mode';
+import { isDevMode } from '@angular/core';
 
 export class HmrOptionBuilder {
   public readonly deferTime: number;
   public readonly autoClearLogs: boolean;
+  public readonly isIvyMode: boolean;
 
-  constructor({ deferTime, autoClearLogs }: NgxsHmrOptions) {
+  constructor({ deferTime, autoClearLogs, isIvyMode }: NgxsHmrOptions) {
     this.deferTime = deferTime || 100;
     this.autoClearLogs = autoClearLogs === undefined ? true : autoClearLogs;
-    this.validateIvyMode();
+    this.isIvyMode = isIvyMode === undefined ? HmrOptionBuilder.isIvy() : isIvyMode;
```

Comment:
Or shorter: `this.ngModule.instance.constructor.hasOwnProperty(ɵNG_MOD_DEF)`

Reply:
`ɵNG_MOD_DEF` - not found for Angular 8

Reply:
Then `.hasOwnProperty('ɵmod')`

Reply:
Sorry :( I try `.hasOwnProperty('ɵmod')`
But it have in JIT and AOT 

`__annotations__` only JIT

Reply:
It checks if Ivy is enabled not if it's JIT or AOT.

Reply:
@arturovt ohh, sorry

Reply:
@arturovt updated